vuu-ui/packages/vuu-shell/src/theme-provider/ThemeProvider.tsx

```diff
@@ -51,24 +65,29 @@ interface ThemeProviderProps {
   children: ReactElement;
   density?: Density;
   theme?: string;
-  applyClassesToChild?: true;
+  themeMode?: ThemeMode;
+  applyClassesTo?: TargetElement;
 }
 
 export const ThemeProvider = ({
   children,
-  density: densityProp,
-  theme: themeProp,
+  // theme: themeProp,
+  themeMode: themeModeProp,
+  density: densityProp
 }: ThemeProviderProps) => {
-  const { theme: inheritedTheme, density: inheritedDensity } =
+  const { density: initialDensity, themeMode: initialThemeMode } =
```

Comment:
we still have the same problem here as discussed previously. Both props and setState can be used by clients to control the state. It needs to be one or the other. 

Thinking this through more carefully, we cannot introduce useState into the provider, it must be managed via props, as in the original version. Inheritance via the nesting of ThemeProviders won't work properly if Providers manage state internally with useState.  

Lets discuss in person